string desiredState
flower m_flower